rfc/0000-configuration-and-shared-files.md

```diff
@@ -0,0 +1,296 @@
+
+#### Precedence
+
+The following order of precedence _must_ be followed when reading configuration data:
+
+1. CLI arguments (if applicable)
+2. Environment variables
+3. Configuration files, in the following order:
+   1. If an explicit configuration file is set, only that file should be read.
+   2. If no explicit configuration file is set, the following files should be read:
+
+      1. Local configuration file
+      2. User configuration file
+
+      The options are merged, with the local configuration file taking precedence over the user configuration file.
+4. Default values
+
+The explicit configuration file may either be set by the `PACT_CONFIG_FILE` environment variable, or by a command line option.
+
+#### Location
+
+To ensure a consistent experience across the Pact ecosystem, the following configuration files are proposed to be used across the Pact ecosystem:
+
+- **User Configuration File**
+
+  Location:
+
+  - On all systems, if the configuration file is explicitly set (either by `PACT_CONFIG_FILE` or a command line option), this location _must_ be used.
+  - Unix and Unix-like systems:
+    - `$XDG_CONFIG_HOME/pact/config.toml` (which defaults to `~/.config/pact/config.toml` if `XDG_CONFIG_HOME` is not set)
+  - Windows:
+    - `%APPDATA%\pact\config.toml`
+  - Fallback:
+    - `~/.pact/config.toml`
+
+  > [!NOTE]
+  >
+  > macOS is considered a Unix-like system for the purposes of this RFC. Should an end-user prefer to use a different location (such as `~/Library/Preferences`), they may do so by setting the `XDG_CONFIG_HOME` environment variable.
+
+  If the operating system cannot be detected, the fallback location _should_ be used.
+
+  If the operating system is detected and the configuration file is found in the fallback location, a warning _should_ be displayed informing the user to relocate the configuration. The exception to this is if the location of the config file is explicitly set, in which case the warning _must not_ be displayed.
+
+  If an end user has multiple user configuration files, it is undefined behaviour.
+
+- **Local Configuration File**
+
+  Location:
+
+  - In the process's working directory, either of:
+    - `pact.toml`
+    - `.pact.toml`
+  - Or looking up the directory tree for the first of the above files.
+
+  If both `pact.toml` and `.pact.toml` are found in the same directory, it is undefined behaviour.
+
+  The parsing of the configuration file _must_ stop at the first file found.
+
+#### Format
+
+The TOML configuration file _must_ be organised into sections (or tables) to group configuration data. Top-level configuration data _must not_ be allowed.
+
+The following tables are proposed:
+
+- `pact`
+
+  To store configuration that is shared across the Pact ecosystem:
+
+  ```toml
+  [pact]
+  plugin_dir = "~/.local/share/pact/plugins"
+  data_home = "~/.local/share/pact"
+  do_not_track = true
+  ```
+
+  The following entries are proposed as part of this RFC:
```

Comment:
Is it only me, or Github format large of the code starting from this line with red background?

Reply:
If you are just reading the diff, it may not have the correct syntax highlighting. I would recommend you read the rendered view (though it does make commenting more difficult).

Reply:
Nothing we can do about this. We can resolve this one.